lib/nahmii-provider.js

```diff
@@ -193,6 +193,29 @@ class NahmiiProvider extends ethers.providers.JsonRpcProvider {
     getAllReceipts() {
         return _nahmii.get(this).get('/trading/receipts');
     }
+
+    /**
+     * Waits for a transaction to be mined, polling every second. 
+     * Rejects if a transaction is mined, but fails to execute, for example in an out of gas scenario.
+     * @see https://docs.ethers.io/ethers.js/html/api-providers.html#transaction-receipts
+     * @param {string} transactionHash
+     * @param {number} [timeout=60] - Seconds to wait before timing out
+     * @returns {Promise<Object>}
+     * @example   
+     * const transactionHash = await wallet.depositEth('1.1', {gasLimit: 200000});
+     * const transactionReceipt = await waitForConfirmation(transactionHash);
+     */
+    waitForConfirmation(transactionHash, timeout=60) {
+        return Promise.retry(() => {
+            process.stderr.write('.');
```

Comment:
Good point. For now, I think it would be ok to remove the writing from the SDK? As you say, if people want it in their application they can add it however they like